### PR TITLE
update heading design-tokens

### PR DESCRIPTION
**Pull request by mcwinter07**
## Why
A support task for the Button epic is to update all heading styles to the new design to create better visual alignment.

## Old Heading values
Tag | font-size (px) | font-size (rem) | line-height (px) | line-height (rem) | font-weight
-- | -- | -- | -- | -- | --
display0  | 72 | 4.5 | 84 | 5.25 | 800
heading1 | 34 | 2.125 | 42 | 2.625 | 700
heading2 | 28 | 1.75 | 36 | 2.25 | 700
heading3 | 22 | 1.375 | 30 | 1.875 | 700
heading4 | 18 | 1.125 | 24 | 1.5 | 600
heading5 | 16 | 1 | 24 | 1.5 | 600
heading6 | 14 | 0.875 | 24 | 1.5 | 600

## New Heading values
Tag | font-size (px) | font-size (rem) | line-height (px) | line-height (rem) | font-weight
-- | -- | -- | -- | -- | --
display0 (cur) | 72 | 4.5 | 84 | 5.25 | 800
heading1 | 36 | 2.25 | 48 | 3 | 500
heading2 | 28 | 1.75 | 40 | 2.5 | 600
heading3 | 20 | 1.25 | 32 | 2 | 600
heading4 | 16 | 1 | 24 | 1.5 | 600
heading5 | 14 | 0.875 | 20 | 1.25 | 600
heading6 | 12 | 0.75 | 16 | 1 | 600

### Tokens updates
- Updated heading design-tokens to new values
  - h1's font-family now be update to `tiempos` instead of inter
- Updated tailwind text-heading class to use heading2 font family
  -  This ensures that all headings that were `inter` will not transition unexpectedly change to `tiempos` for our consumers

### Component updates
There are several component updates required due to the shift in values, especially those use a `heading-6` token or variant. The list of component changes can be found below:

- Avatar typography set to use inter and existing font weight
- `Menu`  v1, v3 and future `Select`'s`ListBoxSection` headings updated to use heading 5 variants to keep visual hierarchy
  - h6 styles were deemed too small
-

### CODE REVIEW

**Comment by mcwinter07** [comment]
Not actually ready for review but doing this to create a branch preview

File: packages/design-tokens/src/js/tokens.ts [hunk 1]
```diff
@@ -224,45 +224,49 @@ export const tokens: TokenStructure = {
       letterSpacing: "0em",
     },
     heading1: {
```
Comment (HeartSquared) [inline packages/design-tokens/src/js/tokens.ts:226]: Potentially worth updating the composable header variant in Heading to use these :)

**Comment by mcwinter07** [comment]
Kicking off a read for review for new chromatic snap shots 🥾💥🏈

**Comment by mcwinter07** [comment]
This is the old typography heading token update. Going to close this PR

**Pull request closed without merging**